### PR TITLE
cp-56: Add chats and chat messages tables

### PR DESCRIPTION
**Pull request by ossavemu**
I have added the tables that chats and chat messages, creating the corresponding migration files. I created a helper to avoid repeating code and updated the default migration to use the helper.

image

image

### CODE REVIEW

File: backend/src/db/migrations/20230215152530_add_users_table.ts
Comment (Xxandder) [inline backend/src/db/migrations/20230215152530_add_users_table.ts:1]: Hi, I think here you need to get rid of TableNames and UserTableColumns enums.  I had the same problem : [here](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/pull/46#discussion_r1301949765)
And the same for other another migration

File: backend/src/db/migrations/20230822123000_add_chat_messages_table.ts
Comment (Xxandder) [inline backend/src/db/migrations/20230822123000_add_chat_messages_table.ts:1]: You also need to add onDelete and onUpdate for foreign columns so messages will be automaticly deleted/updated when a chat or sender is deleted/updated

Comment (Xxandder) [reply backend/src/db/migrations/20230822123000_add_chat_messages_table.ts:1]: Sorry for missing changes blocks in my comments. I just didn't know about how to add it

File: backend/src/db/migrations/20230215152530_add_users_table.ts
Comment (liza-veis) [reply backend/src/db/migrations/20230215152530_add_users_table.ts:1]: Agree

File: backend/src/db/helpers/migration-helper.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { type Knex } from 'knex';
+
+import { CommonTableColumns } from '#libs/enums/enums.js';
+
+function createTableWithCommonColumns(
```
Comment (v0ldemar01) [inline backend/src/db/helpers/migration-helper.ts]: As @Xxandder mentioned, we need to create an independent migrations. So your idea is good, bit not suitable for migrations

File: backend/src/db/migrations/20230215152530_add_users_table.ts [hunk 2]
```diff
@@ -1,35 +1,20 @@
...
 
-const ColumnName = {
-  ID: 'id',
-  EMAIL: 'email',
-  PASSWORD_HASH: 'password_hash',
-  PASSWORD_SALT: 'password_salt',
-  CREATED_AT: 'created_at',
-  UPDATED_AT: 'updated_at',
-};
+const up = createTableWithCommonColumns(
```
Comment (v0ldemar01) [inline backend/src/db/migrations/20230215152530_add_users_table.ts]: It is not a good idea to modify existing migrations

File: backend/src/libs/enums/chat-messages-table-columns.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const ChatMessagesTableColumns = {
+  NAME: 'name',
+  CHAT_ID: 'chat_id',
+  SENDER_ID: 'sender_id',
+  MESSAGE: 'message',
+};
```
Comment (v0ldemar01) [inline backend/src/libs/enums/chat-messages-table-columns.ts]:
```suggestion
} as const;
```

File: backend/src/libs/enums/chats-table-columns.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+const ChatsTableColumns = {
+  NAME: 'name',
+};
```
Comment (v0ldemar01) [inline backend/src/libs/enums/chats-table-columns.ts]:
```suggestion
} as const;
```

File: backend/src/libs/enums/common-table-columns.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+const CommonTableColumns = {
+  ID: 'id',
+  CREATED_AT: 'created_at',
+  UPDATED_AT: 'updated_at',
+};
```
Comment (v0ldemar01) [inline backend/src/libs/enums/common-table-columns.ts]:
```suggestion
} as const;
```

File: backend/src/libs/enums/chats-table-columns.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+const ChatsTableColumns = {
```
Comment (v0ldemar01) [inline backend/src/libs/enums/chats-table-columns.ts]: Enums should be in singular form

File: backend/src/libs/enums/chat-messages-table-columns.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
+const ChatMessagesTableColumns = {
```
Comment (v0ldemar01) [inline backend/src/libs/enums/chat-messages-table-columns.ts]: Enums should be in singular form

File: backend/src/libs/enums/table-names.ts [hunk 8]
```diff
@@ -0,0 +1,7 @@
+const TableNames = {
```
Comment (v0ldemar01) [inline backend/src/libs/enums/table-names.ts]: We already have DatabaseTableName

File: backend/src/libs/enums/user-table-columns.ts [hunk 9]
```diff
@@ -0,0 +1,7 @@
+const UserTableColumns = {
```
Comment (v0ldemar01) [inline backend/src/libs/enums/user-table-columns.ts]: It's not a correct destination for that enum

Comment (v0ldemar01) [reply backend/src/libs/enums/user-table-columns.ts]: And please contact with guys which are have the tasks related to user table columns, maybe they are done with it

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

File: backend/src/libs/enums/table-names.ts (hunk 8 above)
Comment (ossavemu) [reply backend/src/libs/enums/table-names.ts]: I used the cp-14: Adding registry functionality #46 as an example when adding the enum

File: backend/src/libs/enums/database-table-name.enum.ts
Comment (v0ldemar01) [inline backend/src/libs/enums/database-table-name.enum.ts:1]: image

File: backend/src/libs/enums/table-columns/chat-messages-table-column.enum.ts [hunk 10]
```diff
@@ -0,0 +1,10 @@
+const ChatMessagesTableColumn = {
```
Comment (v0ldemar01) [inline backend/src/libs/enums/table-columns/chat-messages-table-column.enum.ts]: Please move to appropriate package

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

**Review by liza-veis — APPROVED** [review]
Good job 🙌

**Review by Nanaier — APPROVED** [review]

**Review by fshabanov — APPROVED** [review]

**Review by v0ldemar01 — APPROVED** [review]

**Review by liza-veis — APPROVED** [review]

**Pull request merged**